addons/crm_iap_lead/models/crm_iap_lead_mining_request.py

```diff
@@ -167,7 +167,15 @@ def _prepare_iap_payload(self):
             payload.update({'company_size_min': self.company_size_min,
                             'company_size_max': self.company_size_max})
         if self.industry_ids:
-            payload['industry_ids'] = self.industry_ids.mapped('reveal_id')
+            # accumulate all reveal_ids (separated by ',') into one list
+            # eg: 3 records with values: "175,176", "177" and "190,191"
+            # will become ['175','176','177','190','191']
+            all_industry_ids = [
+                reveal_id.strip()
+                for reveal_ids in self.mapped('industry_ids.reveal_ids')
+                for reveal_id in reveal_ids.split(',')
+            ]
+            payload['industry_ids'] = all_industry_ids
```

Comment:
Shouldn't we uniquify it ? Would that hvave some side effects ?

Reply:
Used to do `search([('odoo_tag', 'in', data['industry_ids'])])` on the IAP side, so I don't believe it would be an issue.